web/custom-plugins/forecast-of-response/src/components/ReoccurringJob.tsx

```diff
@@ -0,0 +1,155 @@
+
+const Wrapper = styled.div`
+  display: flex;
+  background-color: #ffffff;
+  padding: 1rem;
+  flex-direction: column;
+  height: 300px;
+  width: 400px;
+  justify-content: space-between;
+`
+
+const InputWrapper = styled.div`
+  display: flex;
+  flex-direction: column;
+  height: inherit;
+  justify-content: space-around;
+  margin: 20px;
+`
+
+export function CreateReoccurringJob(props: {
+  close: Function
+  removeJob: Function
+  setCronJob: Function
+  value?: TCronJob | undefined
+}) {
+  const { close, removeJob, setCronJob, value } = props
+  const [schedule, setSchedule] = useState<string | null>(null)
+  const [dateRange, setDateRange] = useState<[Date, Date] | [null, null]>([
+    null,
+    null,
+  ])
+  const [interval, setInterval] = useState<Interval>(Interval.DAILY)
+  const [hour, setHour] = useState<string>('23')
+  const [minute, setMinute] = useState<string>('30')
+
+  useEffect(() => {
+    let newMinute = minute
+    let newHour = hour
+    let dayOfMonth = '*'
+    let month = '*'
+    let dayOfWeek = '*'
+
+    switch (interval) {
+      case Interval.WEEKLY:
+        dayOfMonth = '*'
+        dayOfWeek = '6'
+        break
+      case Interval.MONTHLY:
+        dayOfMonth = '1'
+        break
+    }
+    setSchedule(`${newMinute} ${newHour} ${dayOfMonth} ${month} ${dayOfWeek}`)
+  }, [interval, hour, minute])
+  return (
+    <Wrapper>
+      <h3>Set a schedule for the job</h3>
+      <InputWrapper>
+        <DateRangePicker
+          setDateRange={(dateRange: [Date, Date]) => setDateRange(dateRange)}
+          value={dateRange}
+        />
+        <div
+          style={{
+            flexDirection: 'row',
+            display: 'flex',
+            justifyContent: 'space-between',
+          }}
+        >
+          <div>
+            <Label label="Interval" />
+            <StyledSelect onChange={(e: Event) => setInterval(e.target.value)}>
+              {Object.entries(Interval).map(([key, value]: any) => (
+                <option key={key} value={value}>
+                  {value}
+                </option>
+              ))}
+            </StyledSelect>
+          </div>
+          <div>
+            <Label label="Time" />
+            <StyledSelect
+              onChange={(e: Event) => {
+                const hourAndMinute = e.target.value
+                const [newHour, newMinute] = hourAndMinute.split(':')
+                setMinute(newMinute)
+                setHour(newHour)
+              }}
+            >
+              {generateSelectableTimes().map((value: string, index) => (
+                <option key={index} value={value}>
+                  {value}
+                </option>
+              ))}
+            </StyledSelect>
+          </div>
+        </div>
+      </InputWrapper>
+
+      <div
+        style={{
+          display: 'flex',
+          justifyContent: 'space-around',
+          margin: '10px',
+        }}
+      >
+        <Button color={'secondary'} onClick={() => close()}>
```

Comment:
when testing locally, it was difficult to understand the difference between the close and remove button.